databuilder/query_engines/sqlite.py

```diff
@@ -23,7 +23,47 @@ def date_add_days(self, date, num_days):
         return self.date_add("days", date, num_days)
 
     def date_add_months(self, date, num_months):
-        return self.date_add("months", date, num_months)
+        new_date = self.date_add("months", date, num_months)
+        # In cases of day-of-month overflow, SQLite *usually* rolls over to the first of
```

Comment:
For some reason, it's always our date-related code that ends up needing these comment essays :) But in any case, I think this kind of thing is super valuable and thanks for taking the time to write it.